scripts/bootstrap/buildenv.sh

```diff
@@ -18,6 +18,36 @@
 
 set -o errexit
 
+# Check if all necessary tools are available.
+# List: https://github.com/bazelbuild/bazel/issues/7641#issuecomment-472344261
+for tool in basename cat chmod comm cp dirname find grep ln ls mkdir mktemp \
+            readlink rm sed sort tail touch tr uname unzip which; do
+  if ! which "$tool" >/dev/null; then
```

Comment:
The old code used the `bash`-native `hash` command instead of relying on the external `which(1)`. I think we should stay with that, as in this way we are certain to use the same search information as the shell will later use.

Reply:
Done.